### PR TITLE
RavenDB-14435 Don't stops replication after double dispose

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs [hunk 1]
```diff
@@ -1096,8 +1097,13 @@ private void AddReplicationPerformance(IncomingReplicationStatsAggregator stats)
                 _lastReplicationStats.TryDequeue(out stats);
         }
 
+        private readonly MultipleUseFlag _disposed = new MultipleUseFlag();
+
         public void Dispose()
         {
+            if (_disposed.Raise() == false)
+                return; // already disposed
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs]: Wouldn't it be better to use DisposeOnce? It was created for this purpose.

File: src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs [hunk 2]
```diff
@@ -357,7 +357,7 @@ private void Replicate()
                     if (_parent.DebugWaitAndRunReplicationOnce != null)
                     {
                         _parent.DebugWaitAndRunReplicationOnce.Wait(_cts.Token);
-                        _parent.DebugWaitAndRunReplicationOnce.Reset();
+                        _parent.DebugWaitAndRunReplicationOnce?.Reset();
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:360]: We checked that this is not null 2 lines up?

File: src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs]: done

File: src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs:360]: in the test I do
```
database.ReplicationLoader.DebugWaitAndRunReplicationOnce = null;
mre.Set();
```

**Review by ppekrol — APPROVED** [review]

**Pull request merged**